src/item.cpp

```diff
@@ -4621,16 +4623,55 @@ std::string item::tname( unsigned int quantity, bool with_prefix, unsigned int t
                   contents_item.charges > 1 )
                 ? contents_item.charges
                 : 1;
+
+            // Color second half of container contents depending on perishability (highest perishability of all contents)
+            if( contents_item.is_food() ) {
+                // For comestibles
+				const item_pocket* const parent_pocket = contained_where(contents_item);
+                colorprefix = "<" + string_from_color( contents_item.color_in_inventory() ) + ">";
+                colorprefix.replace( 1, 1, "color" ); // changes <c_cyan> to <color_cyan>
+                if( contents.get_sealed_summary() == item_contents::sealed_summary::all_sealed && parent_pocket->spoil_multiplier() == 0 ) {
+                    colorprefix = "<color_light_blue>";
+                }
+                colorsuffix = "</color>";
+            }
+
             contents_suffix_text = string_format( pgettext( "item name",
                                                   //~ [container item name] " > [inner item  name]"
-                                                  " > %1$s" ),
+                                                  " > %1$s%2$s" ),
+                                                  /* with_contents=false for nested items to prevent excessively long names */
+                                                  colorprefix, contents_item.tname( contents_count, true, 0, false )
                                                   /* with_contents=false for nested items to prevent excessively long names */
-                                                  contents_item.tname( contents_count, true, 0, false ) );
+                                                ) + string_format( "%s", colorsuffix );
+
         } else if( !contents.empty() ) {
-            contents_suffix_text = string_format( npgettext( "item name",
+
+            int worstrot = 0;
+                // If container has multiple food items, take the worst rot state and use that as the color for the "# items" text.
+                // 0 = No food; 1 = Non-perishable (cyan); 2 = Perishable (light cyan); 3 = Old (yellow); 4 = Rotten (brown)
+                for( const item *it : contents.all_items_top( item_pocket::pocket_type::CONTAINER ) ) {
+                    if( worstrot < 4 && it->rotten() ) {
```

Comment:
Why did you remove the check for sealed container here?

Reply:
Is it currently possible to seal a container with multiple items in such a way that it is no longer rotting, and for the container to have multiple "pockets" in which one is sealed and no longer rotting while other pockets are not sealed and actively rotting? 
I removed it due to complexity of handling the above case (and that sealing a container with multiple items in it is not something I think is supported right now (?))

Reply:
I actually would be happy to write sealed (non-rotting) support for multiple items if you have a quick way to generate a sealed + spoil modifier = 0 container with multiple items that are perishable inside so I can test it!

Reply:
I think you can write a unit test that creates one such item and check the name of that item, if you want.